Tests/CareKitEssentialsTests/CareTaskProgressStrategynTests.swift

```diff
@@ -0,0 +1,233 @@
+//
+//  CareTaskProgressStrategyTests.swift
+//  CareTaskProgressStrategyTests
+//
+//  Created by Luis Millan on 7/10/24.
+//  Copyright © 2024 Network Reconnaissance Lab. All rights reserved.
+//
+
+import XCTest
+import CareKitStore
+@testable import CareKitEssentials
+
+final class CareTaskProgressStrategyTests: XCTestCase {
+
+    func testProgressByAveragingOutcomeValuesNoOutcomeNoTarget() async throws {
+        let event = OCKAnyEvent.mock(
+            taskUUID: UUID(),
+            occurrence: 0,
+            hasOutcome: false
+        )
+        // swiftlint:disable:next line_length
+        let progress = CareTaskProgressStrategy<LinearCareTaskProgress>.computeProgressByAveragingOutcomeValues(for: event)
+
+        XCTAssertEqual(progress.value, 0.0, accuracy: 0.0001)
+        XCTAssertNil(progress.goal)
+    }
+
+    func testProgressAveragingSingleOutcomeSingleTarget() async throws {
+        let outcomeValues = [OCKOutcomeValue(10)]
+        let targetValues = [OCKOutcomeValue(20)]
+        let event = OCKAnyEvent.mock(
+            taskUUID: UUID(),
+            occurrence: 0,
+            hasOutcome: true,
+            values: outcomeValues,
+            targetValues: targetValues
+        )
+        // swiftlint:disable:next line_length
+        let progress = CareTaskProgressStrategy<LinearCareTaskProgress>.computeProgressByAveragingOutcomeValues(for: event)
+        XCTAssertEqual(progress.value, 10.0, accuracy: 0.0001)
+        XCTAssertEqual(progress.goal!, 20.0, accuracy: 0.0001)
+    }
+
+    func testProgressAveragingMultipleOutcomesMultipleTargets() async throws {
+        let outcomeValues = [OCKOutcomeValue(10),
+                             OCKOutcomeValue(20),
+                             OCKOutcomeValue(30)]
+        let targetValues = [OCKOutcomeValue(15),
+                            OCKOutcomeValue(25),
+                            OCKOutcomeValue(35)]
+        let event = OCKAnyEvent.mock(
+            taskUUID: UUID(),
+            occurrence: 0,
+            hasOutcome: true,
+            values: outcomeValues,
+            targetValues: targetValues
+        )
+        // swiftlint:disable:next line_length
+        let progress = CareTaskProgressStrategy<LinearCareTaskProgress>.computeProgressByAveragingOutcomeValues(for: event)
+        // swiftlint:disable:next line_length
+        let expectedValue = outcomeValues.map { $0.numberValue?.doubleValue ?? 0.0 }.reduce(0, +) / Double(outcomeValues.count)
```

Comment:
The extra casting shouldn't be needed, you can grab a `Double` value directly. Please make this change to all `$0.numberValue?.doubleValue`:

```suggestion
        let expectedValue = outcomeValues.map { $0.doubleValue ?? 0.0 }.reduce(0, +) / Double(outcomeValues.count)
```